test/feature/middleware_support.feature

```diff
@@ -0,0 +1,60 @@
+Feature: Middleware support
+
+
+  Background:
+    Given a file "routes.coffee" with the content
+      """
+      module.exports = ({ GET }) ->
+        GET '/users', to: 'users#index'
+      """
+
+
+  Scenario Outline: simple middleware support
+    Given a file "controllers/users_controller.coffee" with the content
+      """
+      middle = (req, res, next) ->
+        req.custom = 'foo'
+        next()
+
+      class UsersController
+        middleware:
+          index: middle
+
+        index: (req, res) -> res.end req.custom
+      module.exports = UsersController
+      """
+    And an exprestive app using defaults
+    When making a <REQUEST> request to "<URL>"
+    Then the response body should be "<RESPONSE BODY>"
+
+    Examples:
+      | REQUEST | URL    | RESPONSE BODY |
+      | GET     | /users | foo           |
```

Comment:
This can just be a scenario instead of a scenario outline with one example. Same for below